src/file/FileController.ts

```diff
@@ -228,3 +230,44 @@ export async function downloadAndDecryptDataFile(file: TutanotaFile, fileFacade:
 		return convertToDataFile(file, bytes)
 	}
 }
+
+type DownloadArgs<T> = {
+	tutanotaFiles: TutanotaFile[]
+	downloadAction: (file: TutanotaFile) => Promise<T>
+	processDownloadedFiles: (downloadedFiles: T[]) => Promise<unknown>
+	cleanUp: (downloadedFiles: T[]) => Promise<unknown>
+	progress?: stream<number>
+}
+
+export async function doDownload<T>(args: DownloadArgs<T>) {
```

Comment:
I feel like this could be a place where we could use something like the template method pattern to make the callsites more readable. we only have so many combinations of these args.